docs/Events/listening-to-events.md

```diff
@@ -9,3 +9,71 @@ parent: "Events"
 ---
 
 # Listening to Events
+
+`bpmn-js` uses a general purpose [event bus](../Modules/common-modules/event-bus.md) provided by `diagram-js` to communicate across a diagram instance based on [`events`](./bpmn-events).
+
+## Retrieving the `eventBus` Instance
+
+In order to register a listener for an `event`, the `eventBus` instance needs to be retrieved first. This instance is registered as named diagram service in an internal class called `BaseViewer` and can therefore be retrieved using the `BaseViewer#get` method which is available in the sub-classes [`NavigatedViewer`](tbd), [`Viewer`](tbd) or [`Modeler`](tbd).
+
+Alternatievely the `#on` and `#off` methods can be used in [`NavigatedViewer`](tbd), [`Viewer`](tbd) or [`Modeler`](tbd) to directly register or remove a previously registered event listener.
+
+### Example Usage
+
+Notice that the following snippet demonstrates the retrieval of the `eventBus` instance using a [`Modeler`](tbd). In principle this instance could also be retrieved from an instance of a [`NavigatedViewer`](tbd) or [`Viewer`](tbd) likewise.
+```javascript
+import BpmnJS from "bpmn-js/lib/Modeler";
+
+const modeler = new BpmnJS({
+  container,
+  keyboard: {
+    bindTo: document
+  }
+});
+
+// [...]
+
+const eventBus = modeler.get('eventBus');
+```
+
+
+## Registering an Event Listener
+
+[EventBus](../Modules/common-modules/event-bus.md) provides the [`EventBus#on`](tbd) and [`EventBus#once`](tbd) methods for registering events (see the reference pages for more details).
+
+### Example Usage
+
+Register a listener function with custom priority of `1500` (default is `1000`, where a higher number results in a higher prioity and therefore earlier calling).
+
+Notice that the following snippet could also use the [`EventBus#once`](tbd) method resulting in the listener function only being called once.
+```javascript
+function listener(event) {
+  // access event type
```

Comment:
Maybe only a minor thing. We enforce new lines before comments in our linting styles, cf. https://github.com/bpmn-io/eslint-plugin-bpmn-io/blob/master/configs/recommended.js#L18

Maybe we should enforce in our docs code examples as well

```js
function listener(event) {

  // access event type
  event.type;
}